aten/src/ATen/native/cuda/ReduceOpsKernel.cu

```diff
@@ -41,7 +41,7 @@ void std_var_kernel_impl<at::Half>(TensorIterator& iter, bool unbiased, bool tak
 
 template <typename scalar_t, typename acc_t=scalar_t>
 void prod_kernel_impl(TensorIterator& iter) {
-  gpu_reduce_kernel<scalar_t, scalar_t>(iter, func_wrapper<scalar_t> ([]GPU_LAMBDA(acc_t a, acc_t b) -> acc_t {
+  gpu_reduce_kernel<scalar_t, acc_t>(iter, func_wrapper<scalar_t> ([]GPU_LAMBDA(acc_t a, acc_t b) -> acc_t {
```

Comment:
we should have a test

Reply:
Sure, @nairbv. I might have to do a few more modifications to the `prod_kernel_impl` as well. And also write a unit test, & commit. Thanks!